tests/chainerx_tests/unit_tests/routines_tests/test_linalg.py

```diff
@@ -93,3 +93,117 @@ def test_dot_invalid(is_module, xp, device, a_shape, b_shape, dtype):
         return xp.dot(a, b)
     else:
         return a.dot(b)
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    chainer.testing.product({
+        'shape': [(1, 1), (3, 3), (6, 6)],
+        'in_dtypes': ['float32', 'float64'],
+    })
+))
+class TestSolve(op_utils.NumpyOpTest):
+
+    dodge_nondifferentiable = True
+
+    def setup(self):
+        device = chainerx.get_default_device()
+        if (device.name == 'native:0'
+                and not chainerx.linalg._is_lapack_available()):
+            pytest.skip('LAPACK is not linked to ChainerX')
+
+    def generate_inputs(self):
+        a = numpy.random.random(self.shape).astype(self.in_dtypes)
+        b = numpy.random.random(self.shape[0]).astype(self.in_dtypes)
+        return a, b
+
+    def forward_xp(self, inputs, xp):
+        a, b = inputs
+        out = xp.linalg.solve(a, b)
+        return out,
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'shape': [(2, 3), (3, 2)],
+        'in_dtypes': ['float32', 'float64'],
+    })
+))
+class TestSolveFailing(op_utils.NumpyOpTest):
+
+    forward_accept_errors = (numpy.linalg.LinAlgError,
+                             chainerx.DimensionError)
+
+    def setup(self):
+        device = chainerx.get_default_device()
+        if (device.name == 'native:0'
+                and not chainerx.linalg._is_lapack_available()):
+            pytest.skip('LAPACK is not linked to ChainerX')
+
+    def generate_inputs(self):
+        a = numpy.random.random(self.shape).astype(self.in_dtypes)
+        b = numpy.random.random(self.shape).astype(self.in_dtypes)
+        return a, b
+
+    def forward_xp(self, inputs, xp):
+        a, b = inputs
+        out = xp.linalg.solve(a, b)
+        return out,
+
+
+@op_utils.op_test(['native:0', 'cuda:0'])
+@chainer.testing.parameterize(*(
+    chainer.testing.product({
+        'shape': [(1, 1), (3, 3), (6, 6)],
+        'in_dtypes': ['float32', 'float64'],
+    })
+))
+class TestInverse(op_utils.NumpyOpTest):
+
+    dodge_nondifferentiable = True
+
+    def setup(self):
+        device = chainerx.get_default_device()
+        if (device.name == 'native:0'
+                and not chainerx.linalg._is_lapack_available()):
+            pytest.skip('LAPACK is not linked to ChainerX')
```

Comment:
How about creating a common test base class that does this check for you?

Reply:
Yes, it would be better. I need to do it.

Reply:
Done in [commit](https://github.com/chainer/chainer/pull/7474/commits/a5dc604bb7c2d89f8c4b7b8ef2b0d52abeec57aa).